Add examples from the HIP programming guide (#271)
amd/rocm-examples: HIP-Programming-Guide/CMakeLists.txt

# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

cmake_minimum_required(VERSION 3.21 FATAL_ERROR)
project(HIP-Programming-Guide LANGUAGES CXX)
include(CTest)

include("${CMAKE_CURRENT_LIST_DIR}/../Common/HipPlatform.cmake")
select_gpu_language()
enable_language(${ROCM_EXAMPLES_GPU_LANGUAGE})
select_hip_platform()

add_subdirectory(add_kernel)
if(NOT ${ROCM_EXAMPLES_GPU_LANGUAGE} STREQUAL "CUDA")
    message(STATUS "CUDA not in use, not building block reduction example.")
else()
    add_subdirectory(block_reduction)
endif()
add_subdirectory(complex_math)
add_subdirectory(constant_memory)
add_subdirectory(dynamic_shared_memory)
add_subdirectory(dynamic_unified_memory)
add_subdirectory(explicit_copy)
add_subdirectory(kernel_memory_allocation)
if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
    add_subdirectory(low_precision_float_fp8)
    add_subdirectory(low_precision_float_fp16)
else()

>     message(STATUS "Not building on Linux, not building low-precision floating-point examples.")

Reviewer: ```suggestion
    message(STATUS "Low-precision floating-point examples are available only on Linux.")
```
